### PR TITLE
Remove deprecated API from ConnectorPageSourceProvider

### PR DESCRIPTION
**Pull request by Praveen2112**

### CODE REVIEW

File: presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java [hunk 1]
```diff
@@ -26,19 +26,6 @@ default ConnectorPageSource createPageSource(
             ConnectorSplit split,
```
Comment (kokosing) [inline presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java]: a typo in commit msg and PR title

File: presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java [hunk 2]
```diff
@@ -26,19 +26,6 @@ default ConnectorPageSource createPageSource(
...
             ConnectorTableHandle table,
             List<ColumnHandle> columns)
-    {
-        return createPageSource(transaction, session, split, columns);
-    }
-
-    /**
-     * @param columns columns that should show up in the output page, in this order
-     */
-    @Deprecated
```
Comment (kokosing) [inline presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java]: when it was deprecated?

Comment (Praveen2112) [reply presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java]: IIRC we had one issue where TupleDomain was common for a table scan and we are copying it for all splits so we capture it in TableHandle and pass it to the worker so the size of split can be reduced

**Comment by Praveen2112** [comment]
@kokosing Thanks for pointing it out. Have updated it

File: presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java (hunk 2 above)
Comment (findepi) [reply presto-spi/src/main/java/io/prestosql/spi/connector/ConnectorPageSourceProvider.java]: https://github.com/prestosql/presto/commit/e2e6fd3326d7b15ac40f22c16c237c30428c9a85

**Comment by kokosing** [comment]
Merged, thanks!

**Pull request merged**